### ISSUE TITLE
Tooltip: appendTo not working when targeting elements besides body

### ISSUE THREAD

**Issue by mattb103190** [primefaces/primeng#17243]
### Describe the bug

`appendTo` seems broken for me on the tooltip component when giving a target element that's not the body.

Side note: `autoHide` also doesn't seem to do anything.

### Reason for not contributing a PR

- [X] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://stackblitz.com/edit/github-h6dcpcrk?file=src%2Fapp%2Fapp.component.html

### Environment

Ubuntu

### Angular version

19.0.5

### PrimeNG version

v19

### Node version

20.9.0

### Steps to reproduce the behavior

Add `appendTo` with a target element to a tooltip component, hover.

### Expected behavior

The tooltip should correctly be applied as a child element to the target element, and be removed from the DOM when hovered off.

**Comment by mark7p** [comment]
Hi @mattb103190 I submitted a PR that solves the `appendTo` bug.

> Side note: autoHide also doesn't seem to do anything.

Can you create another issue for `autoHide`? Thanks

**Comment by tbui17** [comment]
I don't think this was resolved entirely with the PR, but it did at least make the tooltip appear. The tooltip doesn't append to the given target. 

Here's a repro on version `19.0.6`
https://stackblitz.com/edit/9sgn3tn9-rnunvwsn?file=src%2Fapp%2Ftooltip-position-demo.html

The issue comes from incomplete handling of the appendTo option. Unfortunately, this issue touches a LOT of other code and I'm hesitant to make these changes right now without some tests in place. There's a Typescript error in the [tests for `Panel`](https://github.com/primefaces/primeng/blob/fd48576f863fa5bdf5d46afbd14959b13ad80475/packages/primeng/src/panel/panel.spec.ts#L4) that's preventing tests from compiling. This might need a bigger PR. There's another issue #17033 caused by the general set of problems here.

- [`getHostOffset`](https://github.com/primefaces/primeng/blob/master/packages/primeng/src/tooltip/tooltip.ts#L517) and `activeElement` need to handle `appendTo` when it's an `ElementRef` or `TemplateRef` respective to their roles in offset and alignment.
- The directional align methods should all reference the `activeElement` but only one does. `getHostOffset` needs to reference `activeElement` since we'd want to perform offset calculations based on the active element (whether it's the element the directive is placed on or the `appendTo` target if applicable).
- The `appendTo` option can be an `ElementRef` or `TemplateRef`. [`appendChild`](https://github.com/primefaces/primeuix/blob/c12c2b3e3ddcc8b102ac5688611ac5654777ec38/packages/utils/src/dom/methods/appendChild.ts) and [`removeChild`](https://github.com/primefaces/primeuix/blob/c12c2b3e3ddcc8b102ac5688611ac5654777ec38/packages/utils/src/dom/methods/removeChild.ts) depend on `toElement` which doesn't handle either of those options (returning `undefined`), which will cause an error in `create` for example when you attempt to append `undefined` in `appendChild`. Handling of `appendTo` across all these cases might need to be factored out into its own function or possibly even a class to make these fixes reliably.
  - [`toElement`](https://github.com/primefaces/primeuix/blob/main/packages/utils/src/dom/methods/toElement.ts) is also in a separate package and depended on by multiple variants of the framework (React, Vue, Angular). Changing this might cause some unwanted changes across many other repositories.

**Comment by mark7p** [comment]
Hi @tbui17 , are you expecting the Tooltip to appear over the target element referred in `appendTo` ? If that's the case, I think there is a misunderstanding on how the `appendTo` API should be used.

The purpose of this API is to control where the tooltip is added in the DOM. In your [repro](https://stackblitz.com/edit/9sgn3tn9-rnunvwsn?file=src%2Fapp%2Ftooltip-position-demo.html), the tooltip is working as expected, it is appended inside the `<button #appendButton>` as a child to that element.

If you really need the Tooltip to behave like it's contained to the target element. Add `position: relative;` to the target element like the example [here](https://stackblitz.com/edit/9sgn3tn9-xrzw7gnj?file=src%2Fapp%2Ftooltip-position-demo.html).

**Comment by tbui17** [comment]
Hi @mark7p, thanks for the clarification and quick response!

If this is the case, then that saves me a lot of work in handling the behavior for `appendTo` while fixing the rendering with `ScrollTop`. The other issues I've mentioned aren't related to this issue; I'll be investigating them separately.
